ios/objective-c/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/Ampli/Ampli.h

```diff
@@ -0,0 +1,607 @@
+/**
+ * Ampli - A strong typed wrapper for your Analytics
+ *
+ * This file is generated by Amplitude.
+ * To update run 'ampli pull obj-c-ampli'
+ *
+ * Required dependencies: Amplitude-iOS^8.6.0
+ * Tracking Plan Version: 0
+ * Build: 1.0.0
+ *
+ * [View Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest)
+ *
+ * [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/obj-c-ampli)
+ */
+#import <Foundation/Foundation.h>
+#import "Amplitude.h"
+
+typedef NSMutableDictionary<NSString *, id> MiddlewareExtra;
+
+typedef NS_ENUM(NSInteger, AmpliEnvironment) {
+  development,
+  production
+};
+
+@interface Event: NSObject
+@property (nonatomic, strong, readonly) NSString * _Nonnull eventType;
+@property (nonatomic, strong, readonly) NSDictionary * _Nullable eventProperties;
+@end
+
+@interface LoadClientOptions: NSObject
+@property (nonatomic, strong, readwrite) NSString * _Nullable apiKey;
+@property (nonatomic, strong, readwrite) Amplitude * _Nullable instance;
++ (instancetype _Nonnull)initWithApiKey: (NSString *_Nonnull) apiKey;
++ (instancetype _Nonnull)initWithAmplitudeInstance: (Amplitude *_Nonnull)instance;
+@end
+
+@interface LoadOptions: NSObject
+@property (nonatomic, readwrite) AmpliEnvironment environment;
+@property (nonatomic, readwrite) BOOL disabled;
+@property (nonatomic, readwrite) LoadClientOptions * _Nullable client;
++ (instancetype _Nonnull)initWithApiKey:(NSString *_Nonnull) apiKey;
++ (instancetype _Nonnull)initWithAmplitudeInstance:(Amplitude *_Nonnull)instance;
++ (instancetype _Nonnull)initWithEnvironment:(AmpliEnvironment)environment;
++ (instancetype _Nonnull)initWithDisabled:(BOOL) disabled;
++ (instancetype _Nonnull)initWithEnvironmentAndDisabled:(AmpliEnvironment)environment disabled:(BOOL) disabled;
++ (instancetype _Nonnull)initWithClient:(LoadClientOptions *_Nonnull)client;
+@end
+
+@interface EventOptions : NSObject
```

Comment:
Builder block here as well. It's likely we will add all HttpV2 fields (optional) in the future which could be tedious to instantiate otherwise.